hera/editor/behavior/DesignBehavior.tsx

```diff
@@ -228,11 +234,48 @@ export default class DesignBehavior {
     return null;
   }
 
+  private putMultiple(
+    vectors: Array<Vector>,
+    state: State,
+    actions: Actions,
+    editor: EditorState,
+  ): StateLike | null {
+    let newState: StateLike | null = null;
+    const players = Array.from(
+      // `PlayerIDs` starts with `0`, which we don't want to include here.
+      new Set([...state.map.active, ...PlayerIDs.slice(1)]),
+    ).slice(0, vectors.length);
+    vectors.forEach((vector, index) => {
+      const currentPlayerIndex = players.indexOf(
+        state.map.getCurrentPlayer().id,
+      );
+      const playerId =
+        players[
+          ((currentPlayerIndex >= 0 ? currentPlayerIndex : 0) + index) %
+            players.length
+        ];
+
+      const tempState = this.put(
+        vector,
+        { ...state, ...newState },
+        actions,
+        editor,
+        playerId,
+      );
+      newState = {
+        ...newState,
+        ...tempState,
+      };
```

Comment:
Currently it'll place all on all possible positions. If for example a tile can't be placed on a mirror cursor because there already is a building there, that position is ignored.

We could change that behavior to not perform any change, if any of the placements is invalid by checking if `tempState.map === undefined` and returning `null` here.

Ignoring invalid placements seems more natural to me though.

Reply:
I agree.